### PR TITLE
test(jupiter): add integration tests for jupiter

### PR DESCRIPTION
**Pull request by jhaeyaert**
Since we were able to build apim, test sdk is now available. This PR adds the integration tests for:

- Jupiter execution mode
- Jupiter with V3 compatibility execution mode

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.5.0-jupiter-integration-tests-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-apikey/2.5.0-jupiter-integration-tests-SNAPSHOT/gravitee-policy-apikey-2.5.0-jupiter-integration-tests-SNAPSHOT.zip)

### CODE REVIEW

File: src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import io.gravitee.apim.gateway.tests.sdk.configuration.GatewayConfigurationBuilder;
+import io.gravitee.definition.model.Api;
+import io.gravitee.definition.model.ExecutionMode;
+import io.gravitee.policy.v3.apikey.ApiKeyPolicyV3IntegrationTest;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class ApiKeyPolicyIntegrationTest extends ApiKeyPolicyV3IntegrationTest {
```
Comment (ytvnr) [inline src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java:27]:
```suggestion
public class ApiKeyPolicyFullJupiterIntegrationTest extends ApiKeyPolicyV3IntegrationTest {
```

File: src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationV3CompatibilityTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import io.gravitee.apim.gateway.tests.sdk.configuration.GatewayConfigurationBuilder;
+import io.gravitee.definition.model.Api;
+import io.gravitee.definition.model.ExecutionMode;
+import io.gravitee.policy.v3.apikey.ApiKeyPolicyV3IntegrationTest;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class ApiKeyPolicyIntegrationV3CompatibilityTest extends ApiKeyPolicyV3IntegrationTest {
```
Comment (ytvnr) [inline src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationV3CompatibilityTest.java:27]:
```suggestion
public class ApiKeyPolicyV3CompatibilityIntegrationTest extends ApiKeyPolicyV3IntegrationTest {
```

**Review by ytvnr — COMMENTED** [review]
maybe we could keep `IntegrationTest` as a suffix and apply the following naming convention ?
- `V3IntegrationTest`: current mode
- `FullJupiterIntegrationTest`: GW and API in Jupiter mode
- `V3CompatibilityIntegrationTest`: GW Jupiter and API in V3

wdyt ?

File: src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java (hunk 1 above)
Comment (jhaeyaert) [reply src/test/java/io/gravitee/policy/apikey/ApiKeyPolicyIntegrationTest.java:27]: As a naming convention we will choose to name the integration test as follow:

- `ApiKeyPolicyIntegrationTest`: full Jupiter mode
- `ApiKeyPolicyV3CompatibilityIntegrationTest`: Jupiter mode with V3 execution compatibility
- `ApiKeyPolicyV3IntegrationTest`: full v3, no Jupiter at all

**Review by ytvnr — APPROVED** [review]

**Pull request merged**